activerecord/lib/active_record/connection_adapters/postgresql/schema_definitions.rb

```diff
@@ -42,7 +42,16 @@ module ColumnMethods
         # a record (as primary keys cannot be +nil+). This might be done via the
         # +SecureRandom.uuid+ method and a +before_save+ callback, for instance.
         def primary_key(name, type = :primary_key, **options)
-          options[:default] = options.fetch(:default, "gen_random_uuid()") if type == :uuid
+          if type == :uuid
+            options[:default] = options.fetch(:default, "gen_random_uuid()")
+          elsif options.delete(:auto_increment) == true && !%i(serial bigserial).include?(type)
+            type = if type == :bigint || options[:limit] == 8
```

Comment:
Actually we treat `type = :integer, limit = 8` as a proxy for `type = :bigint` at https://github.com/rails/rails/blob/v5.0.1.rc1/activerecord/lib/active_record/connection_adapters/postgresql/schema_statements.rb#L649.
But We should handle only `type = :integer, auto_increment = true` case for the above reason.